consensus/module_consensus_state_sync.go

```diff
@@ -22,13 +23,36 @@ func (m *consensusModule) GetNodeAddress() string {
 	return m.nodeAddress
 }
 
-// TODO(#352): Implement this function, currently a placeholder.
-// commitReceivedBlocks commits the blocks received from the blocksReceived channel
+// blockApplicationLoop commits the blocks received from the blocksReceived channel
 // it is intended to be run as a background process
 func (m *consensusModule) blockApplicationLoop() {
-	// runs as a background process in consensus module
-	// listens on the blocksReceived channel
-	// commits the received block
+	for blockResponse := range m.blocksReceived {
+		block := blockResponse.Block
+		maxPersistedHeight, err := m.maxPersistedBlockHeight()
+		if err != nil {
+			m.logger.Err(err).Msg("couldn't query max persisted height")
+			return
+		}
+
```

Comment:
`block.BlockHeader.Height > m.CurrentHeight()`

Does this mean we throw away future blocks? Doesn't this mean we need to re-request them again?

Reply:
yes. Added a todo to consider not throw away future blocks